spek-table-driven-extension/src/main/kotlin/org/jetbrains/spek/table/TableDriven.kt

```diff
@@ -0,0 +1,320 @@
+package org.jetbrains.spek.table
+
+import org.jetbrains.spek.api.dsl.Pending
+import org.jetbrains.spek.api.dsl.Spec
+import org.jetbrains.spek.api.dsl.TestBody
+import org.jetbrains.spek.api.dsl.TestContainer
+
+data class Example1<out P1> internal constructor(val p1: P1)
+
+fun <P1> example(p1: P1): Example1<P1> = Example1(p1)
+fun <P1> Spec.with(vararg examples: Example1<P1>, testContent: TestContainer. (P1) -> Unit) {
+    examples.forEach {
+        val proxyContainer = object : TestContainer by this@with {
```

Comment:
Sorry @FaustXVI forgot to reply. I take back what I said before, I think it should just contain `test` scopes. As you said, it will be hard to find a place to put the information of the current case.